server/src/main/java/org/elasticsearch/index/fielddata/ScriptDocValues.java

```diff
@@ -141,17 +144,27 @@ public int size() {
         }
     }
 
-    public static final class Dates extends ScriptDocValues<ReadableDateTime> {
-        protected static final DeprecationLogger deprecationLogger = new DeprecationLogger(ESLoggerFactory.getLogger(Dates.class));
+    public static final class Dates extends ScriptDocValues<Object> {
```

Comment:
wouldn't is be simpler to have 2 `Dates` classes and keep the type safety instead? I mean I understand we would duplicate the code I still kinda like that better and we can put the conditional in a single place?

Reply:
We don't actually have type safety already, because painless exposes raw ScriptDocValues (because it can't know for a given field name which ScriptDocValues specific class is to be used at compile time). I think it would also detach the relevant code for little gain (having to do this selection of which Dates class to use inside field data code where these are constructed).